### ISSUE TITLE
Start of game hang/lag

### ISSUE THREAD

**Issue by DanielHouston** [ValveSoftware/Dota-2#1971]
#### Your system information

* System information from steam (`Steam` -> `Help` -> `System Information`) in a [gist](https://gist.github.com/): 
https://gist.github.com/DanielHouston/3252564f86c30a1c554c9a3fff0738cb 
* Have you checked for system updates?: [Yes/No]
Fully up to date on Fedora 34 - 5.13.10-200
* Are you using the latest stable video driver available for your system? [Yes/No]
Yes  - Nvidia 470.63.01
* Have you verified the game files?: [Yes/No]
Yes 

#### Please describe your issue in as much detail as possible:
I'm using vulkan, have had no real issues with Dota on fedora for years.
But this month, at the start of games, I am getting upwards of 3 seconds of what appears like hanging or a large "lag" spike. 
I also have some mangohud logs showing the fps dropping to 1 during the hang/lag: https://gist.github.com/DanielHouston/44ae0fbecad8a514c604efd2d8faf48f

Interestingly, this is only when joining my own games, DotaTV seems to behave fine.

I'd be happy to investigate further myself, but not sure where a good place to start is.

#### Steps for reproducing this issue:

1. Start dota with -vulkan
2. Queue and enter a game
3. On loading the game, everything hangs/lags

**Comment by rahenry** [comment]
I've experienced the same

**Comment by VelorumS** [comment]
Similar to https://github.com/ValveSoftware/Dota-2/issues/1970

**Comment by AxelMaure** [comment]
Same issue here. But now i can't even pick a hero.
-Ubuntu  20.04
-Vulkan
pls! im getting an abandon every time a try to play!

**Comment by Saladin1812** [comment]
Yes this issue and issue #1970 are because of the shaders. This is happening since around 7.30.

**Comment by eeyrjmr** [comment]
yup, once 7.30 was release, all UX went downhill.
if you manage to get into a game and if you do not switch desktop's its runs smoothly but..
1) startup takes forever
2) navigation is sluggish

**Comment by guianselmi** [comment]
Similar to what I reported on https://github.com/ValveSoftware/Dota-2/issues/1968

**Comment by alpacalife** [comment]
I have this same issue that came up around the same time. Didn't have it prior. Define lagging as video going to 1FPS and no controls.

Seems to happen at these times:
* Menu/Intro video take forever to load.
* Hero selection lag at initial load, Strategy Time.
* Lag at Verses screen load and initial getting into game.

Once in-game, zero lag.

On Vulkan on Arch 5.13.13-zen1-1-zen w/ mesa 21.2.1-1 
Ryzen 5 3600x
AMD 6800xt


EDIT:

I just tested with launch under OpenGL `-gl` based on #1968. These issues don't seem to happen.
I tested with Steam Shader Cache disabled as per #1970 and issues also went alway when launching with Vulkan `-vulkan`

**Comment by eeyrjmr** [comment]
This is agnostic to nvidia and amd.  
I have been using nvidia for decades and this issue kicked in with the 7.30 patch.
I then replaced my old nvidia card (for other reasons) with an amd 6600xt and the same issue

**Comment by Saladin1812** [comment]
I recommend AMD users to update mesa to 21.2.2 release. It is available in Arch and soon in fedora. You can check it on Fedora bodhi and on Arch it should already be available in your arch mirror. That mesa release fixed the issues.

**Comment by VelorumS** [comment]
@jatin-cbs well, that's not a fix since it's happening on nvidia too.

**Comment by Saladin1812** [comment]
> @jatin-cbs well, that's not a fix since it's happening on nvidia too.

Ya specifying AMD..

**Comment by jatin-cbs** [comment]
However this introduced some other issues

**Comment by romulasry** [comment]
Same issue. Very laggy and can't enter the options menu without absurd waiting time.
